howto_io.md

```diff
@@ -0,0 +1,304 @@
+# How-to: NumPy I/O
+- [1 Reading text files](#1-reading-text-files)
+  * [1.1 Read a CSV or other text file with no missing values](#11-read-a-csvhttpsenwikipediaorgwikicomma-separated_values-or-other-text-file-with-no-missing-values)
+  * [1.2 Read a CSV or other text file with missing values](#12-read-a-csvhttpsenwikipediaorgwikicomma-separated_values-or-other-text-file-with-missing-values)
+    + [1.2.1 Non-whitespace delimiters](#121-non-whitespace-delimiters)
+      - [1.2.1.1 Masked-array output](#1211-masked-array-output)
+      - [1.2.1.2 Array output](#1212-array-output)
+      - [1.2.1.3 Array output, specified fill-in value](#1213-array-output-specified-fill-in-value)
+    + [1.2.2 Whitespace-delimited](#122-whitespace-delimited)
+- [2 Read a file in .npy or .npz format](#2-read-a-file-in-npy-or-npz-format)
+- [3 Write to a file to be read back by NumPy](#3-write-to-a-file-to-be-read-back-by-numpy)
+  * [3.1 Binary](#31-binary)
+  * [3.2 Human-readable](#32-human-readable)
+  * [3.3 Large arrays](#33-large-arrays)
+- [4 Read an arbitrarily formatted binary file ("binary blob")](#4-read-an-arbitrarily-formatted-binary-file-binary-blob)
+- [5 Write or read large arrays](#5-write-or-read-large-arrays)
+- [6 Write files for reading by other (non-NumPy) tools](#6-write-files-for-reading-by-other-non-numpy-tools)
+- [7 Write or read a JSON file](#7-write-or-read-a-json-file)
+- [8 Save/restore using a pickle file](#8-saverestore-using-a-pickle-file)
+- [9 Convert from a pandas DataFrame to a NumPy array](#9-convert-from-a-pandas-dataframe-to-a-numpy-array)
+- [10 Save/restore using `numpy.tofile` and `numpy.fromfile`](#10-saverestore-using-numpytofile-and-numpyfromfile)
+- [11 See all NumPy's I/O routines](#11-see-all-numpys-io-routines)
+
+<a name="1-reading-text-files"></a>
+## 1 Reading text files
+
+<a name="11-read-a-csvhttpsenwikipediaorgwikicomma-separated_values-or-other-text-file-with-no-missing-values"></a>

```

Comment:
A shorter anchor would be better

Reply:
Valid point. If I may, *The Zen of Anchors*:

> An anchor that works is better than a missing anchor.
> An anchor generated by software is better than one generated by hand.
> The look of an anchor is the sound of one hand clapping.

So yes, the anchor length is ridiculous, but since it isn't clear what format this doc will ultimately take, I'd like to set this consideration aside for now.